### PR TITLE
Readd redirecting of stdout and stderr

### LINKED ISSUE #5507: [1.13.2] log4j is having trouble process stdout
~~Since coremods are javascripts running in Nashorn with limited access to classpath, and print() only prints in stdout.~~
Edit: okay, i don't have experience in javascripts.
image

### PR DESCRIPTION
**Pull request by ichttt**
This fixes #5507
This is a modified copy of [the original TracingPrintStream](https://github.com/MinecraftForge/MinecraftForge/blob/1.12.x/src/main/java/net/minecraftforge/fml/common/TracingPrintStream.java), but cleaned up and expanded to also work for primitives that are logged.

### CODE REVIEW

File: src/fmllauncher/java/net/minecraftforge/fml/loading/TracingPrintStream.java [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+    private void log(String s)
+    {
+        logger.info("{}{}", getPrefix(), s);
+    }
+
+    private static String getPrefix()
+    {
+        StackTraceElement[] elems = Thread.currentThread().getStackTrace();
+        StackTraceElement elem = elems[BASE_DEPTH]; // The caller is always at BASE_DEPTH, including this call.
```
Comment (LexManos) [inline src/fmllauncher/java/net/minecraftforge/fml/loading/TracingPrintStream.java]: This is just ASKING for index out of bounds. So protect yourself in here.

Comment (ichttt) [reply src/fmllauncher/java/net/minecraftforge/fml/loading/TracingPrintStream.java]: This should not be possible the way this method is called. I can add some protection though...

Comment (LexManos) [reply src/fmllauncher/java/net/minecraftforge/fml/loading/TracingPrintStream.java]: "This should not be possible" This is just asking for it to be possible.

**Comment by ichttt** [comment]
I don't know why the build on the latest commit failed. It works on my local machine...

**Comment by LexManos** [comment]
ya, there is something wrong with the jenkins or I messed up the JenkinsFile. Cant make the snapshot service. 
But looks good.

**Pull request merged**